### PR TITLE
32 refactor conversion code

### PR DESCRIPTION
**Pull request by nmmull**
* The lib file is pretty much empty at this point. I think this is generally a good thing, though perhaps a bit silly looking for now. I imagine as the project progresses the lib file will contain versions of functions that we want to expose to the user.
* The updates to convert ended up needing to be a bit more complicated than anticipated. Long story short, the `ERROR` submodule for a given module should be reserved for fatal errors and, in particular, it should only be returned, not handled internally. For now the nonfatal errors are handled by a try catch block, and I may eventually port my `LOG` abstraction to this part of the project as well if we find that it is sufficiently useful.
* Please look carefully at the new version of `acopy` and `amap`. I tried to maintain the content of the code while changing the structure, but it might be good to verify that I was successful in this.
* Also note the changes in the signature `CONVERT`.

Generally speaking, I think this is a good baseline for starting testing on individual emails. With everything refactored it should be easier to isolate any necessary changes.

### CODE REVIEW

**Comment by nmmull** [comment]
Also included a unit test file to get everyone started.

File: lib/convert.ml [hunk 1]
```diff
@@ -0,0 +1,388 @@
+open Prelude
+
+module type CONVERT =
```
Comment (bufordrat) [inline lib/convert.ml:3]: So a design question to consider is: do we want different `Conversion_x` modules in their own separate files?  Probably doesn't matter right now.  But I guess my gut says: once each of them gets big enough, maybe they wanna start going in their own files.  

Anyway, not an actual suggestion just yet, given that we only have one module of signature `CONVERT`, but just to say that if we decide to adopt that design, then I would maybe move the `CONVERT` module sig over to `lib.ml`.

**Review by bufordrat — COMMENTED** [review]
Beautiful!  My only suggested change is to have the log go to standard error rather than standard out.

File: lib/convert.ml (hunk 1 above)
Comment (nmmull) [reply lib/convert.ml:3]: Actually that would be a problem, dune will consider that a circularity. If you do want different conversion modules in different files, what I've seen done is to create a separate file, something like `utils.ml`, that just contains all the signature and other globally useful functions.

File: lib/convert.ml [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+            try
+              let src = bhd # field "content-type" in
+              let  trans_lst = Option.default []
+                                 (Configuration.Formats.Dict.find_opt src dict)
+              in
+              if   copy || empty trans_lst
+              then (List.map (transform bhd b) trans_lst) @ [(bhd, `Body b)]
+              else List.map (transform bhd b) trans_lst
+            with _ ->
+              print "no content-type in header\n"; (* TODO: better logging *)
```
[the remark this answers is not in the document]
Comment (nmmull) [reply lib/convert.ml:237]: done.

**Pull request merged**